SparkPost/2web2ui
# FAD-6362 and friends

### FAD-6362: ChangePlanPage assumed that account billing details are already loaded
**Symptom:** Navigating directly to `/account/billing/plan` was broken since it needed billing details that were not available.

**Fix:** Kick off an `/account?include=billing` call on ChangePlanPage

**Testing:**
 - Log in using an account with paid plan and credit card details
 - Navigate directly to `/account/billing/plan`
 - Note that credit card details are pre-filled
 - Complete a plan upgrade

Note: also fixed some other billing tests that were missing mock props.

### FAD-6361: IP Pools form was stuck in edit mode
**Symptom:** Editing the default pool, whose name field is disabled, then hitting "Create IP Pool" cause the form to use the pool state from the previous use of the form. That meant a disabled name field, prefilled with "Default".

**Fix:** Rework IP pools form initial value selector to include create and edit modes using the component's `isNew` prop.

**Testing:**
 - Edit the "Default" pool
 - Create a new pool
 - Note that you can indeed create a new pool and that it's named correctly.

Maybe also worth checking that you can move IPs between pools too.

### FAD-6369: Dedicated IPs form accepts numbers, not +ve integers
**Symptom:* You can request non-integer numbers of dedicated IPs in `/account/billing`. The app copes by using the `floor()` of your input but the validation isn't great (and can give unexpected results if you try).

**Fix:** Added an `integer` validator.

**Testing:**
 - Go to `/account/billing` with an account that allows for dedicated IPs
 - Try to request 1.3, 0.9999999999999999 or -238.1 IPs
 - Be rebuked by the awesome form validation


### FAD-6380: api onboarding curl not working
**Symptom:** The api host had /api/v1/api/v1 and also not adding `\` add end of each line

**Fix:** fixed it

**Testing:**
 - Go to https://app.sparkpost.com/onboarding/api
 - copy curl command
